library/std/src/panicking.rs

```diff
@@ -571,15 +575,23 @@ pub fn begin_panic_handler(info: &PanicInfo<'_>) -> ! {
     }
 
     let loc = info.location().unwrap(); // The current implementation always returns Some
+    let source = info.source();
     let msg = info.message().unwrap(); // The current implementation always returns Some
     crate::sys_common::backtrace::__rust_end_short_backtrace(move || {
         if let Some(msg) = msg.as_str() {
-            rust_panic_with_hook(&mut StrPanicPayload(msg), info.message(), loc, info.can_unwind());
+            rust_panic_with_hook(
+                &mut StrPanicPayload(msg),
+                info.message(),
+                loc,
+                source,
```

Comment:
The message and can_unwind fields are also taken directly, without storing them in a variable:
```suggestion
                info.source(),
```
Or is there a reason that doesn't work here?